File: MARCH 24/31 March/Closest Neighbour in BST gfg.cpp
class Solution {
  public:
    int findMaxForN(Node* root, int n) {
        // code here
        if(root==nullptr)
            return -1;
        
        else if(root->key==n)
            return n;
        else if(root->key<n)
            return max(root->key, findMaxForN(root->right, n));
        else if(root->key>n)
            return findMaxForN(root->left, n);
    }
};

File: MAY 2025/30 May/Closest Neighbour in BST gfg.java
class Solution {
    public int findMaxFork(Node root, int k) {
        // code here.
        if(root == null)
            return -1;
            
        if(root.data<=k)
            return Math.max(root.data, findMaxFork(root.right, k));
        else
            return findMaxFork(root.left, k);
    }
}
